shared/fs/header/popup-container.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+import * as Types from '../../constants/types/fs'
+import {DropdownPopupMenu} from './popup'
+import {connect, type TypedState} from '../../util/container'
+import {type RouteProps} from '../../route-tree/render-route'
+
+type OwnProps = RouteProps<
+  {
+    isTeamPath: boolean,
+    items: Array<Types.PathBreadcrumbItem>,
+    onOpenBreadcrumb: (path: string) => void,
+  },
+  {}
+>
+
+const mapStateToProps = (stateProps: TypedState, {routeProps}: OwnProps) => ({
```

Comment:
`stateProps` is not used here. Why does `DropdownPopupMenu` needs to be connected? It seems it only needs `routeProps` from the parent.

Reply:
I thought it needs to be connected to get `routeProps`

Reply:
@songgao is correct here
ownProps is just the props
so you can use mapProps from recompose if you want to transform it
you only need to connect to get access to the store, or if you want to dispatch actions

Reply:
Hmm I thought that was passed in from `fs/routes.js`. `ownProps` should have nothing to do with connecting if I understand correctly.

Reply:
Cool, I'll make a PR to refactor that down.